MovieApp/MovieApp/App/LocalDataSource/UserDefaultsDataSource.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+class UserDefaultsDataSource: UserDefaultsDataSourceProtocol {
+    
+    static let shared: UserDefaultsDataSource = UserDefaultsDataSource()
+    
+    private let favoritesUDKey = "favorites"
+    
+    var favorites: [Int] {
```

Comment:
Since you're always using `userDefaultsDataSource.favorites.contains`, you might want to replace this with `func isFavorite(movieId: Int) -> Bool` and do the `contains` logic in there.

Reply:
[Here](https://github.com/lihter/Movie-App/pull/38/commits/cd9b26a2e35cb7418ace1f07855e235f95125719#diff-dfe654e503e62b61d6e962c2eec0634be5d54ff8715d6c83602c1572cd599349R31)